ERCS/erc-7765.md

```diff
@@ -0,0 +1,170 @@
+        uint256 _privilegeId
+    );
+
+    /// @notice This function exercise a specific privilege of a token.
+    /// @dev Throws if `_privilegeId` is not a valid privilegeId.
+    /// @param _to  the address to benefit from the privilege.
+    /// @param _tokenId  the NFT tokenID.
+    /// @param _privilegeId  the ID of the privileges.
+    /// @param _data  extra data passed in for extra message or future extension.
+    function exercisePrivilege(
+        address _to,
+        uint256 _tokenId,
+        uint256 _privilegeId,
+        bytes calldata _data
+    ) external;
+
+    /// @notice This function is to check whether a specific privilege of a token can be exercised.
+    /// @dev Throws if `_privilegeId` is not a valid privilegeId.
+    /// @param _to  the address to benefit from the privilege.
+    /// @param _tokenId  the NFT tokenID.
+    /// @param _privilegeId  the ID of the privileges.
+    function isExercisable(
+        address _to,
+        uint256 _tokenId,
+        uint256 _privilegeId
+    ) external view returns (bool _exercisable);
+
+    /// @notice This function is to check whether a specific privilege of a token has been exercised.
+    /// @dev Throws if `_privilegeId` is not a valid privilegeId.
+    /// @param _to  the address to benefit from the privilege.
+    /// @param _tokenId  the NFT tokenID.
+    /// @param _privilegeId  the ID of the privileges.
+    function isExercised(
+        address _to,
+        uint256 _tokenId,
+        uint256 _privilegeId
+    ) external view returns (bool _exercised);
+
+    /// @notice This function is to list all privilegeIds of a token.
+    /// @param _tokenId  the NFT tokenID.
+    function getPrivilegeIds(
+        uint256 _tokenId
+    ) external view returns (uint256[] memory privilegeIds);
+
+}
+```
+
+The function `exercisePrivilege` performs the exercise action to a specific privilege of a token. If succeeds, it is expected to emit a `PrivilegeExercised` event.
+
+The function `getPrivilegeIds` provides a way to manage the binding relationship between NFTs and privilegeIds.
+
+The **metadata extension** is OPTIONAL for [EIP-7765](./eip-7765.md) smart contracts. This allows your smart contract to be interrogated for its details about the privileges which your NFTs carry.
+
+```
+pragma solidity >=0.7.0 <0.9.0;
+
+/// @title ERC-7765 Privileged Non-Fungible Tokens Tied To Real World Assets, optional metadata extension
+/// @dev See https://eips.ethereum.org/EIPS/eip-7765
+interface IERC7765Metadata /* is IERC7765 */ {
+
+    /// @notice A distinct Uniform Resource Identifier (URI) for a given privilegeId.
+    /// @dev Throws if `_privilegeId` is not a valid privilegeId. URIs are defined in RFC
+    ///  3986. The URI may point to a JSON file that conforms to the "ERC-7765
+    ///  Metadata JSON Schema".
+    function privilegeURI(uint256 _privilegeId) external view returns (string memory);
+
+}
+```
+
+This is the “EIP-7765 Metadata JSON Schema” referenced above.
```

Comment:
You should probably say what format this schema is in. It looks like JSON Schema?